### PR TITLE
add CPU PoW lib

### PR DESCRIPTION
**Pull request by auxten**
add CPU PoW lib and UT

### CODE REVIEW

**Review by zeqing-guo — APPROVED** [review]

File: crypto/hash/hash.go [hunk 1]
```diff
@@ -93,6 +94,24 @@ func (hash *Hash) IsEqual(target *Hash) bool {
...
 }
 
+// Difficulty returns the leading Zero **bit** count of Hash in binary.
+//  return -1 indicate the Hash pointer is nil
+func (hash *Hash) Difficulty() (difficulty int) {
+	if hash == nil {
+		return -1
+	}
+
+	for i, _ := range *hash {
```
Comment (covenantdb) [inline crypto/hash/hash.go]: **[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
should omit 2nd value from range; this loop is equivalent to `for i := range ...`

File: pow/cpuminer/miner.go [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+func NewCPUMiner(quit chan struct{}) *CPUMiner {
+	return &CPUMiner{quit: quit}
+}
+
+// HashBlock calculate the hash of MiningBlock
+func (miner *CPUMiner) HashBlock(data []byte, nonce big.Int) hash.Hash {
+	return hash.DoubleHashH(append(data, nonce.Bytes()...))
+}
+
+// CalclateBlockNonce find nonce make HashBlock() match the MiningBlock Difficulty from the startNonce
```
Comment (covenantdb) [inline pow/cpuminer/miner.go]: **[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
comment on exported method CPUMiner.CalculateBlockNonce should be of the form "CalculateBlockNonce ..."

File: crypto/hash/hash.go (hunk 1 above)
Comment (covenantdb) [inline crypto/hash/hash.go]: **[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
should omit 2nd value from range; this loop is equivalent to `for i := range ...`

File: pow/cpuminer/miner.go (hunk 2 above)
Comment (covenantdb) [inline pow/cpuminer/miner.go]: **[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
comment on exported method CPUMiner.CalculateBlockNonce should be of the form "CalculateBlockNonce ..."

File: crypto/hash/hash.go (hunk 1 above)
Comment (covenantdb) [inline crypto/hash/hash.go]: **[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
should omit 2nd value from range; this loop is equivalent to `for i := range ...`

File: pow/cpuminer/miner.go (hunk 2 above)
Comment (covenantdb) [inline pow/cpuminer/miner.go]: **[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
comment on exported method CPUMiner.CalculateBlockNonce should be of the form "CalculateBlockNonce ..."

**Pull request merged**